### ISSUE TITLE
Test loss evaluated on only 8 samples

### ISSUE THREAD

**Issue by tileb1** [wei-mao-2019/LearnTrajDep#12]
Hello,

If I am not mistaken, I see that the test data loader for each act only has 1 batch of size 8. The test loss (subject 5) is thus only evaluated on 1 single batch of size 8 (both for angle and 3D space) for each act.

The variable opt.test_batch is not reflected in the code because the H36motion dataset only contains 8 sequences.

Can you explain the reason behind this? The test loss seems to have a high variance for different runs, I suppose having such a small test set could explain this.

Thanks in advance,

Tim

**Comment by wei-mao-2019** [comment]
Hi Tim,

We follow the same evaluation protocol as previous methods (e.g. conSeq2Seq and Res. Sup.) which is testing with 8 random subsequences per action. To make the comparision fair, we evaluate with the exact same 8 subsequences as previous methods.

Yes, the small testing set will lead to high variance which has already been studyed in  a latter publication  "Modeling Human Motion with Quaternion-based Neural Networks". In this work, the authors suggest to use 256 random subsequences per action instead which makes more sense than only using 8.

Cheers
Wei

**Comment by tileb1** [comment]
Ok, thank you for the clarification! Given the high variance of the accuracies, how did you generate the tables in the readme and in the paper? Is the pre-trained model chosen randomly as well as test_run_1, and test_run_2?

I am trying to compare your results with another version I wrote and I am trying to compare it as fairly as possible.

Thank you in advance!

Best,

Tim

**Comment by wei-mao-2019** [comment]
Hi Tim,

They are all evaluate the exact same sequences which is randomly chosen with same random seed. Our baselines are also evaluated on the same sequences.

You can find the related method here:
https://github.com/wei-mao-2019/LearnTrajDep/blob/15c2bc5aafdbfae4e92f353ae19f00f7b12aea8c/utils/data_utils.py#L793
This method is originally from https://github.com/una-dinosauria/human-motion-prediction/blob/master/src/seq2seq_model.py.

Thank you 
Cheers
Wei
